### PR TITLE
Alter replication number for table in one command

### PR DESCRIPTION
**Pull request by xy720**
#2299 #3389 
This cl add new command to set replication number of table in one time.

```
alter table test_tbl set ("replication_num" = "3");
```

It changes replication num of a table that have only one partition.

and

```
alter table test_tbl set ("default.replication_num" = "3");
```

It changes default replication num of the specified table.

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 1]
```diff
@@ -5177,8 +5177,31 @@ public void replayModifyTableProperty(short opCode, ModifyTablePropertyOperation
     }
 
     // The caller need to hold the db write lock
-    public void modifyTableReplicationNum(Database db, OlapTable table, Map<String, String> properties) {
+    public void modifyTableReplicationNum(Database db, OlapTable table, Map<String, String> properties) throws DdlException {
         Preconditions.checkArgument(db.isWriteLockHeldByCurrentThread());
+        short newReplicationNum = (short) -1;
+        try {
+            newReplicationNum = Short.valueOf(properties.get(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM));
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java:5184]: This should be done in `analyze()` of `ModifyTablePropertyClause.java`.
And you can call `PropertyAnalyzer.analyzeReplicationNum()`.

**Review by morningman — CHANGES_REQUESTED** [review]
`alter table test_tbl set ("default.replication_num" = "3");` is an already exist statement, which is used to set the "default" replication num of a table, but not set the real replication number of each partition. So you'd better not changing its behavior.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 1 above)
Comment (xy720) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: ok.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -5176,8 +5176,50 @@ public void replayModifyTableProperty(short opCode, ModifyTablePropertyOperation
         }
     }
 
+    /**
+     * Set replication number for a table that only have one partition.
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java:5180]:
```suggestion
     * Set replication number for unpartitioned table
```

**Review by morningman — CHANGES_REQUESTED** [review]
Please update the document of alter table

**Review by morningman — COMMENTED** [review]
Add unit test for this

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**